test/test.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+set -eu
+
+. ./build.state
+cat build.state
+
+echo "Test: Update diff > 0"
+[ $DIFF -gt 0 ]
+
+echo "Test: Update diff < 10MB"
+[ $DIFF -lt $(( 10 * 1024 * 1024 )) ]
```

Comment:
don't see the need for this file.

Reply:
removed https://github.com/startersclan/docker-sourceservers/pull/114/commits/b9c303c7e1eea9a0a0da3588190db605b0edbdd5

Reply:
i'd think there should be a simple test that the update does not redownload the game from scratch. Such a test may be in `build.sh` directly, or be in `test.sh`. 

Thinking about it, if the image size grows significantly the update, the pipeline should fail and the image should not be pushed.

Reply:
yes but i think the above test is but a workaround not knowing the cause for entire re-downloads. With #113 significant grows are unlikely to happen on `update` on locally built base images but which could occur on `update` on pulled (older) images as I describe in https://github.com/startersclan/docker-sourceservers/pull/114#issuecomment-1455067748. 

this does raise the question of the need for a `hlds-test-update` for executing `update` on pulled images which could return a different result.

Reply:
yes. this is simply a unit test to identify issues such as #113. 

As for whether a real pull of `:latest` for `update` results in a large diff, we would never know, especially if the `:latest` image might be so old that `steamcmd` is a significantly older version that results in issues uncaught by the unit test (where `steamcmd` is the same version).

Reply:
not just `steamcmd` though the binaries are removed in the clean-up step, but the `appcache` could be valid but expired which could require some kind of re-download.

Reply:
yes, that may happen for `:latest` pulls, and for that there's no way to know, but like you mentioned is very unlikely.

so this services to ensure the `update` Dockerfile works as intended.

Reply:
yes, that may happen for `:latest` pulls, and for that there's no way to know, but as discussed it is very unlikely.

so this test serves to ensure the `update` Dockerfile works as intended.

Reply:
#113 would have been identified by this unit test. The reason there was #113 was identified was because of this PR's artifact.

Reply:
The reason #113 was identified was because of the artifact produced by this PR.

Reply:
i think it's better to minimize on needless tests especially this one that isn't actually a unit test at all. with #113 we've deduced the cause for re-downloads which should no longer occur for `hlds`.